### PR TITLE
Add wflow handler

### PR DESCRIPTION
**Pull request by sisyphusSmiling**
Closes: #134 

## Description

- Enables a path for FLOW to move between Cadence & EVM via the VM bridge. On the way to EVM, FLOW is wrapped as and deposited as WFLOW to the receiver. On the way to Cadence, the WFLOW is unwrapped and returned as `@FlowToken.Vault`.
- Anyone who wishes to move FLOW as native FLOW can leverage the existing EVM interface supporting Cadence -> EVM via `coa.deposit()` and `EVMAddress.deposit()` and Cadence <- EVM via `coa.withdraw()`
- This new behavior is updated from the current state which is that FLOW can be moved to EVM, but not from EVM. Adding to the fragmented UX, WFLOW is currently blocked from being bridged to Cadence as to not confuse things on the Cadence side with unnecessary representations of FLOW (which can already be treated like any other `FungibleToken.Vault`).
- The new route means that platforms like Flow Port & Flow Wallet can add functionality enabling users to define whether they want to bridge FLOW from Cadence as native FLOW or WFLOW. Or alternatively, users might be able to bridge FLOW into Cadence from either their EVM FLOW or WFLOW balances, aggregating either into their Cadence FlowToken Vault.

______

For contributor use:

- [x] Targeted PR against `main` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-evm-bridge/blob/master/CONTRIBUTING.md#styleguides).
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc [hunk 1]
```diff
@@ -191,14 +192,229 @@ access(all) contract FlowEVMBridgeHandlers {
...
+
+        /// Returns whether the Handler is enabled
+        access(all) view fun isEnabled(): Bool {
+            return self.enabled
+        }
+        /// Returns the Cadence type handled by the Handler, nil if not set
+        access(all) view fun getTargetType(): Type? {
+            return self.targetType
+        }
+        /// Returns the EVM address handled by the Handler, nil if not set
```
Comment (nvdtf) [inline cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc:224]: cannot be nil

File: cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc [hunk 2]
```diff
@@ -191,14 +192,229 @@ access(all) contract FlowEVMBridgeHandlers {
...
+        /// Sets the target type for the handler
+        access(FlowEVMBridgeHandlerInterfaces.Admin)
+        fun setMinter(_ minter: @{FlowEVMBridgeHandlerInterfaces.TokenMinter}) {
+            panic("WFLOWTokenHandler does not utilize a minter")
+        }
+
+        /// Enables the handler for request handling. The
+        access(FlowEVMBridgeHandlerInterfaces.Admin)
+        fun enableBridging() {
+            self.enabled = true
```
Comment (nvdtf) [inline cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc:406]: do we need a disable function too?

File: cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc [hunk 3]
```diff
@@ -191,14 +192,229 @@ access(all) contract FlowEVMBridgeHandlers {
...
+    /// Given the native interface on EVM.CadenceOwnedAccount and EVM.EVMAddress to move FLOW between Cadence and EVM,
+    /// this handler treats requests to bridge FLOW as WFLOW as a special case.
+    ///
+    access(all) resource WFLOWTokenHandler : FlowEVMBridgeHandlerInterfaces.TokenHandler {
+        /// Flag determining if request handling is enabled
+        access(self) var enabled: Bool
+        /// The Cadence Type this handler fulfills requests for. This field is optional in the event the Cadence token
+        /// contract address is not yet known but the EVM Address must still be filtered via Handler to prevent the
+        /// contract from being onboarded otherwise.
+        access(self) var targetType: Type
```
[the remark this answers is not in the document]
Comment (sisyphusSmiling) [reply cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc:206]: It's not nil in this instance as we're setting the type on init given the specific use case here. I'll update the comment.

File: cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc (hunk 1 above)
Comment (sisyphusSmiling) [reply cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc:224]: The function signature here is a manifestation of `FlowEVMBridgeHandlerInterfaces.HandlerInfo` interface conformance.

File: cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc (hunk 2 above)
Comment (sisyphusSmiling) [reply cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc:406]: I suppose we could add that here, though FlowEVMBridgeConfig allows pausing on a Type basis as well

File: cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc (hunk 3 above)
[the remark this answers is not in the document]
Comment (sisyphusSmiling) [reply cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc:211]: Addressed in [769a6a1](https://github.com/onflow/flow-evm-bridge/pull/137/commits/769a6a111c2787e5940596d4f3b07042d42d4968)

File: cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc (hunk 2 above)
Comment (sisyphusSmiling) [reply cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc:411]: Added in [769a6a1](https://github.com/onflow/flow-evm-bridge/pull/137/commits/769a6a111c2787e5940596d4f3b07042d42d4968)

**Pull request merged**